homeassistant/components/modbus/binary_sensor.py

```diff
@@ -84,14 +83,13 @@ async def async_setup_platform(
             CONF_NAME: "no name",
             CONF_BINARY_SENSORS: config[CONF_INPUTS],
         }
-        config = None
 
     for entry in discovery_info[CONF_BINARY_SENSORS]:
         if CONF_HUB in entry:
             # from old config!
-            hub: ModbusHub = hass.data[MODBUS_DOMAIN][entry[CONF_HUB]]
+            hub = hass.data[MODBUS_DOMAIN][entry[CONF_HUB]]
```

Comment:
Right now it will have `Any` type. To make it `ModbusHub` just add it once.
```suggestion
            hub: ModbusHub = hass.data[MODBUS_DOMAIN][entry[CONF_HUB]]
```

Reply:
If you put ": ModbusHub" in one of the places, many people will wonder why not in the other place, by not having it in neither place it is consistent, and thus easier to read.

Reply:
But with less type checks. You can also like i mentioned before cast it to the proper type.
```
hub = cast(ModbusHub, hass.data[MODBUS_DOMAIN][entry[CONF_HUB]])
```

With that you get correct type info, but a bit less readable.

Reply:
If anything I would use the cast() to have consistency. But please remember there is no demand in this project to use strong typing, it is a recommendation. If you look at the code you will see that we use types in a lot of places, but not in all, and the possibility to choose is a big advantage of Python.

This PR is not about making strong typing in Modbus, to do that I would have to change quite a lot more places, the purpose of this PR is that modbus is no longer excluded from the mypy check, and that is reached.